cmd/libs/go2idl/go-to-protobuf/package.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+)
+
+func newProtobufPackage(packagePath, packageName string) *protobufPackage {
+	pkg := &protobufPackage{
+		PackageName: packageName,
+		PackagePath: packagePath,
```

Comment:
Agreed.  Proto is easier since it won't change the compilation state, but
the subsequent generation should purge before parse at a minimum.  We can
also look into generating a codec outside of the package.

On Nov 2, 2015, at 7:59 PM, Daniel Smith <notifications@github.com> wrote:

In cmd/libs/go2idl/go-to-protobuf/package.go
<https://github.com/kubernetes/kubernetes/pull/16649#discussion_r43705760>:

> +
> +import (
> +	"fmt"
> +	"os"
> +	"path/filepath"
> +	"sort"
> +	"strings"
> +
> +	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
> +	"k8s.io/kubernetes/cmd/libs/go2idl/types"
> +)
> +
> +func newProtobufPackage(packagePath, packageName string) *protobufPackage {
> +	pkg := &protobufPackage{
> +		PackageName: packageName,
> +		PackagePath: packagePath,

I see. I don't feel too strongly about the proto files. We need to be
careful with the generated functions, like maybe deleting generated files
before running the generator. We don't want to get into today's situation
where we can't run the generator because it doesn't compile.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16649/files#r43705760>.